vm-virtio/src/lib.rs

```diff
@@ -186,4 +186,6 @@ pub enum Error {
     FailedSignalingDriver(io::Error),
     VhostUserUpdateMemory(vhost_user::Error),
     EventfdError(io::Error),
+    SetShmRegionsNotImplemented,
+    SetShmRegionsNotSupported,
```

Comment:
Not sure having two errors is that helpful.

Reply:
I wanted to make the difference between devices like `virtio-blk` (which strictly doesn't implement `set_shm_regions()`) and device like virtio-fs which implements it but where we could end up in a situation where it's not authorized to set the shm regions because no cache has been created.

Reply:
I understand the difference, just not sure the value it adds. It's a programming error to call that function if it's not got it set. As such an assertion would be bet best on the case where it's not implemented rather than an error.

Reply:
Done!